### PR TITLE
convert mouse clicks to driver calls

### PR DESCRIPTION
**Pull request by zgontea**
Jan Gontarek 228737
Franciszek Karwowski 228765

### CODE REVIEW

File: src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java [hunk 1]
```diff
@@ -97,6 +99,18 @@ private static void setupLogger(Application application) {
 		application.addComponentMenuElement(Logger.class, "OFF logging", (ActionEvent e) -> logger.setLevel(Level.OFF));
 	}
 
+	private static void setMouseDrawer(Application application) {
+		application.getFreePanel().addMouseListener(new MouseInputAdapter() {
+			@Override
+			public void mouseClicked(java.awt.event.MouseEvent e) {
+				int correctedX = e.getX() - 268;
+				int correctedY = e.getY() - 227;
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java]: How did you get those values? 
They are not arbitrary constants so they should be calculated

File: src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java [hunk 2]
```diff
@@ -97,6 +99,18 @@ private static void setupLogger(Application application) {
 		application.addComponentMenuElement(Logger.class, "OFF logging", (ActionEvent e) -> logger.setLevel(Level.OFF));
 	}
 
+	private static void setMouseDrawer(Application application) {
+		application.getFreePanel().addMouseListener(new MouseInputAdapter() {
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java]: Please extract this adapter to a separate file.

File: src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java [hunk 3]
```diff
@@ -97,6 +99,18 @@ private static void setupLogger(Application application) {
...
+			@Override
+			public void mouseClicked(java.awt.event.MouseEvent e) {
+				int correctedX = e.getX() - 268;
+				int correctedY = e.getY() - 227;
+				
+				DriverFeature.getDriverManager().getCurrentDriver().operateTo(correctedX, correctedY);
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java]: It would be nice to provide a way to use setPosition (maybe right-click?)

**Review by kowallus — CHANGES_REQUESTED** [review]
Please clean the code.

**Comment by zgontea** [comment]
Requested changes applied 🗻

File: src/test/java/edu/kis/powp/jobs2d/events/MouseListener.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        int correctedX = e.getX() - drawAreaWidth / 2;
+        int correctedY = e.getY() - drawAreaHeight / 2;
+
+        switch (e.getButton()) {
+            case LEFT_BUTTON:
+                DriverFeature.getDriverManager().getCurrentDriver().operateTo(correctedX, correctedY);
+                break;
+        
+            case RIGHT_BUTTON:
+                DriverFeature.getDriverManager().getCurrentDriver().setPosition(correctedX, correctedY);
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/events/MouseListener.java]: Duplicated code: DriverFeature.getDriverManager().getCurrentDriver()

File: src/test/java/edu/kis/powp/jobs2d/events/MouseListener.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    public void mouseClicked(java.awt.event.MouseEvent e) {
+        drawAreaWidth = drawArea.getWidth();
+        drawAreaHeight = drawArea.getHeight();
+
+        int correctedX = e.getX() - drawAreaWidth / 2;
+        int correctedY = e.getY() - drawAreaHeight / 2;
+
+        switch (e.getButton()) {
+            case LEFT_BUTTON:
+                DriverFeature.getDriverManager().getCurrentDriver().operateTo(correctedX, correctedY);
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/events/MouseListener.java]: Try remove coupling with DriverFeature.

File: src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java [hunk 6]
```diff
@@ -100,15 +99,7 @@ private static void setupLogger(Application application) {
...
-		application.getFreePanel().addMouseListener(new MouseInputAdapter() {
-			@Override
-			public void mouseClicked(java.awt.event.MouseEvent e) {
-				int correctedX = e.getX() - 268;
-				int correctedY = e.getY() - 227;
-				
-				DriverFeature.getDriverManager().getCurrentDriver().operateTo(correctedX, correctedY);
-			};
-		});
+		application.getFreePanel().addMouseListener(new MouseListener(application.getFreePanel()));
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java]: amazing!!! Please make it simpler. Try to remove application.getFreePanel() code duplication.

**Review by kowallus — CHANGES_REQUESTED** [review]
One more time

**Review by kowallus — APPROVED** [review]

**Comment by kowallus** [comment]
Unfortunately, PR went to the wrong branch!

**Pull request merged**